Stack.js

```diff
@@ -0,0 +1,166 @@
+class Stack {
+    constructor() {
+        this.items = [];
+    }
+
+
+    // Pushes an element onto the stack (numbers only)
+    push(element) {
+        this.items.push(element);
+    }
+
+
+    // Remove the last element from the stack and return it if exists
+    pop() {
+        if (this.isEmpty()) {
+            return 'Stack is Empty';
+        } else {
+            return this.items.pop();
+        }
+    }
+
+
+    // Remove the first element from the stack and return it if exists
+    trim() {
+        if (this.isEmpty()) {
+            return 'Stack is Empty';
+        } else {
+            return this.items.shift();
+        }
+    }
+
+
+    // Checks if the stack is empty
+    isEmpty() {
+        if (this.items.length == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    // Return stack size
+    size() {
+        return this.items.length;
+    }
+
+
+    // Returns the top element of the stack if it exists
+    peek() {
+        if (this.isEmpty()) {
+            return 'Stack is Empty';
+        } else {
+            return this.items[this.size() - 1];
+        }
+    }
+
+
+    // Returns a space-separated string that contains all stack elements
+    displayStack(stack) {
+        if (this.isEmpty()) {
+            return 'Stack is Empty';
+        } else {
+            return 'Stack Elements: [ ' + this.items.toString() + ' ]';
+        }
+    }
+
+
+    // Return a new stack in reverse order
+    reverse_stack(stack) {
+        const reversedStack = new Stack();
+        while (!stack.isEmpty()) {
+            reversedStack.push(stack.pop());
+        }
+        return reversedStack;
+    }
+
+
+    // Return the max element in the stack (return -1 if it does not exists)
+    max() {
+        if (this.isEmpty()) {
+            return -1;
+        } else {
+            const initialValue = this.items[0];
+            return this.items.reduce((max, current) => {
+                return current > max ? current : max;
+            }, initialValue);
+        }
+    }
+
+
+    // Return the min element in the stack (return -1 if it does not exists)
+    min() {
+        if (this.isEmpty()) {
+            return -1;
+        } else {
+            const initialValue = this.items[0];
+            return this.items.reduce((min, current) => {
+                return current < min ? current : min;
+            }, initialValue);
+        }
+    }
+
+
+    // Remove all the elements from the stack
+    clear() {
+        while (!this.isEmpty()) {
```

Comment:
this works, but not performance efficient, please find an easier and quicker way to remove all the elements